StackExchange.Redis/StackExchange/Redis/RedisDatabase.cs

```diff
@@ -1579,6 +1579,330 @@ public IEnumerable<SortedSetEntry> SortedSetScan(RedisKey key, RedisValue patter
+        {
+            var msg = GetStreamClaimMessage(key,
+                consumerGroup,
+                claimingConsumer,
+                minIdleTimeInMs,
+                messageIds,
+                returnJustIds: false,
+                flags: flags);
+
+            return ExecuteSync(msg, ResultProcessor.SingleStream);
+        }
+
+        public Task<RedisStreamEntry[]> StreamClaimMessagesAsync(RedisKey key, RedisValue consumerGroup, RedisValue claimingConsumer, long minIdleTimeInMs, RedisValue[] messageIds, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetStreamClaimMessage(key,
+                consumerGroup,
+                claimingConsumer,
+                minIdleTimeInMs,
+                messageIds,
+                returnJustIds: false,
+                flags: flags);
+
+            return ExecuteAsync(msg, ResultProcessor.SingleStream);
+        }
+
+        public RedisValue[] StreamClaimMessagesReturningIds(RedisKey key, RedisValue consumerGroup, RedisValue claimingConsumer, long minIdleTimeInMs, RedisValue[] messageIds, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetStreamClaimMessage(key,
+                consumerGroup,
+                claimingConsumer,
+                minIdleTimeInMs,
+                messageIds,
+                returnJustIds: true,
+                flags: flags);
+
+            return ExecuteSync(msg, ResultProcessor.RedisValueArray);
+        }
+
+        public Task<RedisValue[]> StreamClaimMessagesReturningIdsAsync(RedisKey key, RedisValue consumerGroup, RedisValue claimingConsumer, long minIdleTimeInMs, RedisValue[] messageIds, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = GetStreamClaimMessage(key,
+                consumerGroup,
+                claimingConsumer,
+                minIdleTimeInMs,
+                messageIds,
+                returnJustIds: true,
+                flags: flags);
+
+            return ExecuteAsync(msg, ResultProcessor.RedisValueArray);
+        }
+
+        public bool StreamCreateConsumerGroup(RedisKey key, RedisValue groupName, RedisValue readFrom, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = Message.Create(Database,
+                flags,
+                RedisCommand.XGROUP,
+                new RedisValue[4]
+                {
+                    StreamConstants.Create,
+                    key.AsRedisValue(),
+                    groupName,
+                    readFrom
+                });
+
+            return ExecuteSync(msg, ResultProcessor.Boolean);
+        }
+
+        public Task<bool> StreamCreateConsumerGroupAsync(RedisKey key, RedisValue groupName, RedisValue readFrom, CommandFlags flags = CommandFlags.None)
+        {
+            var msg = Message.Create(Database,
+                flags,
+                RedisCommand.XGROUP,
+                key,
+                groupName,
+                readFrom);
```

Comment:
Shouldn't this match non-async?

Reply:
Indeed it should. That's what I get for not testing the Async versions yet...